src/network/graphql/__mocks__/dbmodels/mockTransaction.js

```diff
@@ -0,0 +1,54 @@
+import { TransactionStatus } from 'constants';
+
+/*
+* DbModel for Transaction mocks.
+*/
+export default class MockTransaction {
```

Comment:
Why do you need a `MockTransaction` model? Is this in reference to the `resetMockTransactionList()` change?

Let me explain this better...

- The `PendingTxSnackbar` queries real live data based on a tx's status, if its `PENDING` it will show up.
- For testing you have a mock DB, and you are creating these txs with PENDING status right?
- So... when you want to emulate the txs being processed and are no longer in `PENDING`, what do you need to do? You just need to change the status of all the txs from `PENDING` -> `SUCCESS/FAIL`.
- You should be creating a helper function in `mockData.js`. This is your mock DB server right? It does the "insert" into the DB, so therefore you should have a function that does "update" to it as well.
- Not sure the whole logic of creating this MockTransaction class... Move your `setSuccess()` and `setFail()` helper functions to `mockData.js` and have your `resetMockTransactionList()` call this for every tx in your mock DB.

Reply:
@dwalintukan 
Hi deric thank you for review my pr. I think this could be a better way for testing.
These models designed to have the same behavior with the object that a graphql request should return, but can be with more helpful feature, like: 
* data validation (to avoid making wrong callback data and wasting time because of these typo)
* auto filling (fill default data to blank field to make it more like a real callback data, also less copy & paste)
* command chain design pattern (In future we may also have to test topic and oracle, and when mocking topic/oracle we may also need to change related oracles/txs data. if we just use manually generated object for them, we also need to maintain all of these changes one by one, but with command chain we can easily connect related topic/oracle/txs together, then we change one, the other can also be updated automatically )
For only transaction it surely not necessary, but I think it may become helpful when we start testing oracles and topics. I'm glad to know your opinion on this ideas.

Reply:
And currently this model is used like this:

tx1 = mockAddTransaction(1, { type: TransactionType.APPROVE_CREATE_EVENT });
theFunctionWeWantToTestWithPending();
// when we want to set it to success
tx1.setSuccess();
theFunctionWeWantToTestWithSuccess();

Reply:
@ippudo 
I have no issue with having these helper methods that are useful for testing (setSuccess, setFail, etc).

But, the reason I do not want to have `MockTransaction` classes is for the reason of overhead of future changes. If we were using Typescript and we had inheritance, I would say definitely do the `MockTransaction` classes by extending the `Transaction` class. But we are using JS so there is no inheritance.

So say we do have the `Transaction` and `MockTransaction` class. I can "almost" guarantee later down the road that if we have to change the `Transaction` class (say adding a new field), that someone will miss it adding it to the `MockTransaction` class. I have seen this countless times in my career. So, adding the `MockTransaction` class adds this extra overhead where we have to change the class in two places vs one and gives another vector of possibly introducing a bug.

That being said, I would like to keep the testing on the original models. And when you need something specific for testing, create a helper function, e.g:
```
function mockAddTransaction(args) {
  const tx = new Transaction({
    type = '' // One of: TransactionType
    txid = '' // Transaction ID assigned by the blockchain
    status = '' // One of: TransactionStatus
    createdTime = '' // UNIX timestamp when Transaction was created
    // etc, etc, set your default values here
  });
  Object.assign(tx, args); // overwrite any default values with the ones passed in
}

function mockSetTxStatus(tx, status) {
  tx.status = status;
}

function setAllTxsSuccess(txs) {
  _.each(txs, tx => tx.status = 'SUCCESS';
}
```

Reply:
@ippudo 
We are using the UI Transaction class throughout the entire UI. I understand that you are trying to match the backend structure, but every time we call `queryAllTransactions` in the UI we are always mapping it to the UI Transaction model: `const txs = map(result, (tx) => new Transaction(tx));`. Don't think there are any cases where we are using the raw Transaction callback data without mapping it to the Transaction class.

Even in `PendingTxsSnackbarStore`, which is the subject of your tests, you are testing against the UI Transaction model. If you want to validate the backend data that should be validated on the server-side tests. In this case, we are validating the UI front end Transactions.

Reply:
@dwalintukan 
Maybe that is where the problem comes. We currently map the result to the UI model after each `queryAllTransactions()` called, not in `queryAllTransactions()`. And the function I'm mocking is also `queryAllTransactions()`, so if I use UI TX models as callback data, we will use these UI TX models to generate another UI TX model.